src/test/java/com/gui/HomePageTest.java

```diff
@@ -4,40 +4,52 @@
 import com.gui.pages.desktop.HomePage;
 import com.gui.pages.desktop.ProductsListPage;
 import org.testng.Assert;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 
 public class HomePageTest extends AbstractTest {
 
-    @Test
-    public void test(){
-        HomePage homePage =new HomePage(getDriver());
-        ProductsListPage productsListPage = homePage.searchKeyWord("shirt");
-        productsListPage.printAllItems();
-    }
 
-    @Test(groups = "group1")
+    @Test(groups = "group1", dataProvider = "currencyData")
     public void testCurrencyChange() {
```

Comment:
you must use the parameters in the test method signature and inside the test logic.